docs/modules/ROOT/pages/design/cloudevents.adoc

```diff
@@ -0,0 +1,125 @@
+= Cloud Event Delivery Pipeline
+== Overview
+
+KubeArchive uses Cloud Events that are generated by a Knative Eventing ApiServerSource to discover when resources on
+the kubernetes cluster have changed. These Cloud Events contain the json representation of the Kubernetes resource that
+has changed. The KubeArchive Sink (sink) receives these messages and determines, using CEL expression from the
+KubeArchiveConfig, if the data contained in the Cloud Event should be written to the database and potentially deleted
+from the cluster. While the sink is processing a Cloud Event, it is possible at many points for an error to occur,
+especially when the sink has to make a request of the network, particularly, to the Kubernetes API or the database. This
+page proposes how KubeArchive set up its Cloud Event delivery pipeline and how it should consume Cloud Events to achieve
+the following goals:
+
+* KubeArchive receives all Cloud Events from the Knative ApiServerSource
+* Cloud Events sent to KubeArchive can be replayed
+* Cloud Events received by KubeArchive, that cannot be processed successfully, are reported back to the sender as
+  delivery failure
+* Cloud Events sent to the sink will attempt to be redelivered if the initial delivery fails
+* Cloud Events that could not be delivered are replayed to KubeArchive automatically when KubeArchive becomes available
+  again
```

Comment:
I think we can generalize and compact these, what about:

```suggestion
* Resource updates can be rejected by KubeArchive if a problem happens during processing
* Resource updates are sent again to KubeArchive if it does not provide a response back
```

Reply:
There's a section that explains the different HTTP error codes that the sink can respond with and how they should be treated. I don't really view the HTTP error codes as part of the high level goals, but more as required implementation details based on how Knative Eventing handles sending events over HTTP